File: go/platform.go
package wgrok

import (
	"fmt"
	"net/http"
)

// PlatformSendMessage sends a text message via the specified platform.
func PlatformSendMessage(platform, token, target, text string, client *http.Client) (map[string]interface{}, error) {
	switch platform {
	case "webex":
		return SendMessage(token, target, text, client)
	case "slack":
		return SendSlackMessage(token, target, text, client)
	case "discord":
		return SendDiscordMessage(token, target, text, client)
	case "irc":
		return SendIRCMessage(token, target, text)
	default:
		return nil, fmt.Errorf("unsupported platform: %s", platform)
	}
}

// PlatformSendCard sends a message with card/rich content via the specified platform.
func PlatformSendCard(platform, token, target, text string, card interface{}, client *http.Client) (map[string]interface{}, error) {
	switch platform {
	case "webex":
		return SendCard(token, target, text, card, client)
	case "slack":
		return SendSlackCard(token, target, text, card, client)
	case "discord":
		return SendDiscordCard(token, target, text, card, client)
	case "irc":
		return SendIRCCard(token, target, text, card)
	default:
		return nil, fmt.Errorf("unsupported platform: %s", platform)
	}
}

// PlatformSendMessageToRoom sends a text message to a room via the specified platform.
func PlatformSendMessageToRoom(platform, token, roomID, text string, client *http.Client) (map[string]interface{}, error) {
	if platform == "webex" {
		return SendMessageToRoom(token, roomID, text, client)
	}
	return nil, fmt.Errorf("room-based send not supported for platform: %s", platform)
}

// PlatformSendCardToRoom sends a message with card/rich content to a room via the specified platform.
func PlatformSendCardToRoom(platform, token, roomID, text string, card interface{}, client *http.Client) (map[string]interface{}, error) {
	if platform == "webex" {
		return SendCardToRoom(token, roomID, text, card, client)
	}
	return nil, fmt.Errorf("room-based send not supported for platform: %s", platform)
}

File: rust/src/platform.rs
use reqwest::Client;
use serde_json::Value;

use crate::discord;
use crate::irc;
use crate::slack;
use crate::webex;

pub async fn platform_send_message(
    platform: &str,
    token: &str,
    target: &str,
    text: &str,
    client: &Client,
) -> Result<Value, String> {
    match platform {
        "webex" => webex::send_message(token, target, text, client).await,
        "slack" => slack::send_message(token, target, text, client).await,
        "discord" => discord::send_message(token, target, text, client).await,
        "irc" => irc::send_message(token, target, text).await,
        _ => Err(format!("Unsupported platform: {}", platform)),
    }
}

pub async fn platform_send_card(
    platform: &str,
    token: &str,
    target: &str,
    text: &str,
    card: &Value,
    client: &Client,
) -> Result<Value, String> {
    match platform {
        "webex" => webex::send_card(token, target, text, card, client).await,
        "slack" => slack::send_card(token, target, text, card, client).await,
        "discord" => discord::send_card(token, target, text, card, client).await,
        "irc" => irc::send_card(token, target, text, card).await,
        _ => Err(format!("Unsupported platform: {}", platform)),
    }
}

pub async fn platform_send_message_to_room(
    platform: &str,
    token: &str,
    room_id: &str,
    text: &str,
    client: &Client,
) -> Result<Value, String> {
    match platform {
        "webex" => webex::send_message_to_room(token, room_id, text, client).await,
        _ => Err(format!("room-based send not supported for platform: {}", platform)),
    }
}

pub async fn platform_send_card_to_room(
    platform: &str,
    token: &str,
    room_id: &str,
    text: &str,
    card: &Value,
    client: &Client,
) -> Result<Value, String> {
    match platform {
        "webex" => webex::send_card_to_room(token, room_id, text, card, client).await,
        _ => Err(format!("room-based send not supported for platform: {}", platform)),
    }
}
